e2e_test/batch/transaction/same_session.slt

```diff
@@ -0,0 +1,20 @@
+statement ok
+create table t (id int primary key);
+
+# we don't use flush and test in the same session whether delete and insert overlap.
+statement ok
+delete from t;
```

Comment:
Note that as a batch task, `Delete` only returns when `Scan` finishes. So at this point `DELETE` always yield no `Delete` records downstream. I guess even without this PR, the tests can always succeed.

Reply:
Actually, not, the main branch could fail sometimes with this test. Because we don't ensure these 2 transactions were sent to the same DmlExecutor before.

Reply:
Interesting. But the DELETE transaction should be empty.🤔

Reply:
Oh, I missed something in this test. I repeatedly run delete + insert in my local environment, so the e2e test should be insert + (wait 1s) + delete + insert.